data/covering_grammar/lib/error_analysis.py

```diff
@@ -19,51 +19,54 @@
 
 import argparse
 
-import prettytable  # type: ignore
+import prettytable
 import pynini
+from pynini.lib import rewrite
 
 
 def main(args: argparse.Namespace) -> None:
-    cg_fst = pynini.string_file(args.cg_path).closure().optimize()
-    rulematch_predmatch = 0
-    rulematch_pred_notmatch = 0
-    not_rulematch_predmatch = 0
-    not_rulematch_pred_notmatch = 0
-    total_records = 0
-    with open(args.test_path, "r") as source:
-        for line in source:
-            total_records += 1
-            parts = line.split("\t")
-            lg = parts[0].rstrip()
-            act = parts[1].replace(" ", "").replace(".", "").rstrip()
-            pred_pron = parts[2].replace(" ", "").replace(".", "").rstrip()
-            # TODO: use Pynini's rewrite module here.
-            lattice = lg @ cg_fst @ pred_pron
-            lattice.project("output")
-            if lattice.start() == pynini.NO_STATE_ID:
-                if act == pred_pron:
-                    not_rulematch_predmatch += 1
+    with pynini.default_token_type("utf8"):
+        cg_fst = pynini.string_file(args.cg_path).closure().optimize()
+        rulematch_predmatch = 0
+        rulematch_pred_notmatch = 0
+        not_rulematch_predmatch = 0
+        not_rulematch_pred_notmatch = 0
+        total_records = 0
+        with open(args.test_path, "r") as source:
+            for line in source:
+                total_records += 1
+                ortho, hypo_p, gold_p = line.rstrip().split("\t", 2)
+                hypo_p = hypo_p.replace(" ", "")
+                gold_p = gold_p.replace(" ", "")
+                if rewrite.matches(ortho, hypo_p, cg_fst):
+                    if gold_p == hypo_p:
+                        rulematch_predmatch += 1
+                    else:
+                        rulematch_pred_notmatch += 1
                 else:
```

Comment:
Instead of doing:

```python
if A:
   if B:
     ...
   else:
     ...
else:
  if C:
    ...
  else:
    ...
```

it is (IMO) a bit simpler to just write:

```python
if A:
   if B:
     ...
   else:
     ...
elif: C:
  ...
else:
  ...
```

as was done earlier.

Reply:
Addressed.